lib/core/queue.js

```diff
@@ -127,7 +127,13 @@ class CommandQueue extends EventEmitter {
 
     err = err || this.tree.returnError;
 
+    // only used in hooks
     this.emit('queue:finished', err);
```

Comment:
This was required to be executed before checking if the `currentResult` is resolved because some hooks are only resolved after the `queue:finished` event is fired and not firing it before the below condition was resulting in a deadlock where the `currentResult` won't be settled until the hooks are resolved and the hooks won't be resolved until the `currentResult` is settled and the `queue:finished` event is fired after it.